imagenameparser/api/image_name.py

```diff
@@ -0,0 +1,849 @@
+        return self._image_info['prodbase']
+
+    @property
+    def product_major(self):
+        """Get the value of the 'major_version' regex match"""
+        return self._image_info['major_version']
+
+    @property
+    def product_minor(self):
+        """Get the value of the 'minor_version' regex match"""
+        min_ver = self._image_info['minor_version']
+        return min_ver
+
+    @property
+    def product_minor_int(self):
+        """Determine the product minor version (SPX or .X)"""
+        if self.product_minor is None:
+            return 0
+
+        return int(self.product_minor.lower().replace('sp', ''))
+
+    @property
+    def product_version(self):
+        """Determine the product version"""
+        joiner = ''
+        version = []
+        result = None
+        version.append(self.product_major)
+        if (self.is_sle_server or self.is_hpc or self.is_lasso
+                or self.is_sles_serv or self.is_suma):
+            joiner = '-'
+        elif (self.is_caasp or self.is_sumaserver or self.is_sumaproxy
+                or self.is_leap or self.is_micro):
+            joiner = '-'
+        elif self.is_opensuse:
+            joiner = '-'
+
+        if self.product_minor:
+            version.append(self.product_minor)
+
+        if any(version):
+            result = joiner.join(version)
+
+        return result
+
+    @property
+    def has_product_version(self):
+        """ Check if product_version is set"""
+        return self.product_version is not None
+
+    @property
+    def product_version_string(self):
+        """Set the product version string"""
+        if not self.has_product_version:
+            return ""
+        return self.product_version
+
+    @property
+    def product_version_dashed(self):
+        """Get the product version using a '-' as seperator"""
+        return self.product_version_string.replace('.', '-')
+
+    @property
+    def product_version_dash_lower(self):
+        """Get the lower case product version using a '-' as seperator"""
+        return self.product_version_string.replace('.', '-').lower()
+
+    @property
+    def product_version_lower(self):
+        """Get the lower case product version"""
+        return self.product_version_string.lower()
+
+    @property
+    def product_version_spaced(self):
+        """Get the product version using a ' ' as seperator"""
+        return self.product_version_string.replace('-', ' ')
+
+    # Determine how to maintain these mappings best. maybe in a config file?
+    _SUMA_DISTRO = {
+        # Keys should be the value of self.product_version
+        "4.0": "15-SP1",
+        "4.1": "15-SP2",
+        "4.2": "15-SP3",
+        "4.3": "15-SP4"
+    }
+    _MICRO_DISTRO = {
+        # Keys should be the value of self.product_version
+        "5.0": "15-SP2",
+        "5.1": "15-SP3",
+        "5.2": "15-SP3"
+    }
+    _MIRROR_HOST = {
+        "IBS": "ibs-mirror.prv.suse.net",
```

Comment:
This cannot be public and has to be pulled out into configuration. No SUSE internal host information in the world. Of course it's in the PR and that is very unfortunate, but we cannot commit this to the main branch.

Reply:
Re-base cleans up history.